src/nyc_trees/apps/event/views.py

```diff
@@ -429,3 +445,26 @@ def increase_rsvp_limit(request, event_slug):
     return {
         'max_attendees': event.max_attendees
     }
+
+
+@transaction.atomic
+def event_email_unsubscribe(request, event_slug, token):
+    event = get_object_or_404(Event, group=request.group, slug=event_slug)
+
+    try:
+        username, signed_value = token.split(':', 1)
+    except ValueError:
+        # Invalid token
+        return {}
+
+    user = get_object_or_404(User, username=username)
+
+    if not user.is_valid_token(token):
+        # Invalid token
+        return {}
+
+    EventRegistration.objects.filter(event=event, user=user) \
+                             .update(opt_in_emails=False)
+
+    # Redirect to Emails tab on user profile page.
+    return redirect(reverse('user_profile_settings') + '#emails-pane')
```

Comment:
If I'm not logged in, after clicking the unsubscribe link I get sent to the app and it successfully unsubscribes me, but I don't get any notification -- I just get sent to the login page. After I login, I get sent to `/user/settings`, but not the Emails tab.

This makes it look like I need to login to unsubscribe (even though I don't). I think that maybe we need a different redirect for when the user is not logged in.